orttraining/orttraining/python/training/orttrainer_options.py

```diff
@@ -689,20 +710,34 @@ def _check_is_callable(field, value, error):
                 'min': 0,
                 'default': 0
             },
-            'propagate_cast_ops_level': {
-                'type': 'integer',
-                'min': -1,
-                'default': -1
-            },
-            'propagate_cast_ops_allow': {
-                'type': 'list',
-                'schema': {'type': 'string'},
-                'default': []
-            },
             'allow_layer_norm_mod_precision': {
                 'type': 'boolean',
                 'default': False
             },
+            'propagate_cast_ops_config': {
+                'type': 'dict',
+                'default_setter': lambda _: {},
+                'required': False,
+                'schema': {
+                    'strategy': {
+                        'type': 'propagate_cast_ops_strategy',
+                        'nullable': True,
+                        'min': PropagateCastOpsStrategy.INSERT_AND_REDUCE,
```

Comment:
>min [](http://example.com/codeflow?start=25&length=3)

@thiagocrepaldi - is this right to have min/max here ?

Reply:
It seems in order to use min/max here I need to use py::arithmetic() attribute on the corresponding py::enum_. This is probably unnecessary.